### DISCUSSION TITLE
Lottie Export Option

### DISCUSSION THREAD

**Discussion by YasinC2** [friction2d/friction#114] [Ideas]
Any plans to add [Lottie](https://airbnb.io/lottie/) export option?
I know that most of the effects are for rendering and they are not useful for SVG or even Lottie. but SVG output does not support even basic animation features, including sub-path, which is very useful for animation/motion design for website.
Apparently, Synfig and Glaxnimate programs have the possibility of Lottie export, but Friction has a much better UX and UI.

![Introduction_00_sm - Copy](https://github.com/friction2d/friction/assets/44544053/be558dac-9c45-42b1-a66c-afd8e6212e05)

**Comment by rodlie** [comment]
Hi,

There are currently no plans on my end to add Lottie support. The primary reason is that I don't use it, SVG is good enough for my needs.

Never say never, but currently my resources are limited.

---
As far as I know Glaxinmate has good Lottie support (not tried).

**Reply by YasinC2** [reply]
All good, thanks.

**Reply by IcedQuinn** [reply]
Glaxnimate is also adding/added support for Rive which seems to be somewhere between a Lottie competitor and an attempt to revive Flash.

**Reply by rodlie** [reply]
> Glaxnimate is also adding/added support for Rive which seems to be somewhere between a Lottie competitor and an attempt to revive Flash.

Glaxnimate is probably your best (OSS) alternative if support for Lottie and other web player are needed.

Friction will not provide this, it will only support plain SVG's that work on anything, not something that needs third-party software or JS.

**Reply by irwant** [reply]
in good news.  SVG animation exported from Friction can be converted into file Lottie using Glaxnimate.

**Comment by rodlie** [comment]
Some clarifications on Friction and web animations.

Friction jumped on the horse SVG+SMIL for web animations, if that was the best option only the future may tell. This is a tried and tested format for raster and vector animation on the web and "just works". There are alternatives (or the other way around) like SVG+CSS and SVG+JS(+CSS) and other solutions that requires a player, like Lottie and Rive.

All of them have their pros and cons, and I will not go into them here.

I will rather focus on Friction's place in this "ecosystem" of web animations. I want Friction to be the "Inkscape for web animation" and support as many SVG (animation) features as possible, we are far from there yet, but at some point ...

For now we use SVG+SMIL and we are happy with that and will introduce many new features (even interactive) in the future. Now, you might have heard that SVG+SMIL is deprecated. That is partly correct, Google Chrome tried to deprecate it in the past due to low usage, this has been reverted as the usage of SVG+SMIL has become much more wide spread.

The depreciation of SVG+SMIL will at some point happen, but not until a better alternative can replace it. And that alternative is most likely SVG+CSS. Friction will of course adapt to this change, but that is a discussion for a later date.

**Reply by B4uti4github** [reply]
and OpenFL?  i know that in this software, is very similar to how use flash but is missing things for be compatible

**Comment by pgilfernandez** [comment]
Hi @YasinC2, @irwant, @IcedQuinn, @B4uti4github ,

If you are still interested in Friction adding a Lottie exporter, have a look at the following PR which gives you more information about the topic (already added features, discussion, etc...).
**To try a Friction build with Lottie export already working** check the following links (they might get old if we tweak or add new features so it would be better to go to the PR itself):

- linux: https://github.com/friction2d/friction/actions/runs/27066554928/artifacts/7455827716
- windows: https://github.com/friction2d/friction/actions/runs/27066554926/artifacts/7455781164
- macOS: https://github.com/friction2d/friction/actions/runs/27066554925/artifacts/7455927473

Cheers

**Reply by irwant** [reply]
its better than import the svg animation into Glaxnimate or Lottie Creator.

**Reply by irwant** [reply]
<img width="1878" height="145" alt="image" src="https://github.com/user-attachments/assets/be8a99b0-e796-4346-9beb-91f955aeca28" />
I think this is unnecessary keyframe. all the path have keyframe after exported to lottie

**Reply by pgilfernandez** [reply]
> I think this is unnecessary keyframe. all the path have keyframe after exported to lottie

I've fixed it, try it out:
- Windows: https://github.com/friction2d/friction/actions/runs/27402592069/artifacts/7586303051
- Linux: https://github.com/friction2d/friction/actions/runs/27402592068/artifacts/7586379820
- macOS: https://github.com/friction2d/friction/actions/runs/27402592076/artifacts/7590009594

BTW, if you find more issues, please, report them in the main Lottie PR page: https://github.com/friction2d/friction/pull/775

Thanks!

**Comment by pgilfernandez** [comment]
> but SVG output does not support even basic animation features, including sub-path, which is very useful

BTW, rereading @YasinC2 initial post I realized that he was wrong as SMIL animated SVGs do support sub-path animation and Friction is very capable to do it using "Path Effects / Sub-path" 😉

**Reply by YasinC2** [reply]
Interesting! Good to know. 🙂

Hmm, I wonder why I couldn’t get an export from it. Is this a new feature, or was it there back then?
I haven’t worked with web animation or Friction since around that time.

By the way, nice work adding Lottie export to Friction. I haven’t tried it yet, but it’s great to have this capability in Friction as well. Hopefully we’ll see it included in future releases of the app.

**Reply by pgilfernandez** [reply]
> Hmm, I wonder why I couldn’t get an export from it. Is this a new feature, or was it there back then?
I haven’t worked with web animation or Friction since around that time.

It's been there since a lot of time but maybe you didn't set it op correctly, you must add the effect to the "path effect" (also called "path base") or it won't work when exporting to SVG.
If you apply "sub-path" to other categories (outline, outline base, ...) It may work inside Friction but it won't export to SVG correctly 😉 

> By the way, nice work adding Lottie export to Friction. I haven’t tried it yet, but it’s great to have this capability in Friction as well. Hopefully we’ll see it included in future releases of the app.

It could happen for v1.1 as a plugin but there is still some time for that... in the meantime you could work with the test builds of this specific feature 😉
